packages/docs/docs/getting-started.md

```diff
@@ -183,3 +183,34 @@ $ grep -i 'starting modelcall' packages/ai-jsx/llmx.log | yarn workspace ai-jsx
   }
 ],
 ```
+
+## Beyond Text: Image Generation
+
+You are not restricted by text models. Let's create an image for the story as well:
+
+```tsx title="index.tsx"
+function StoryWithImage() {
+  const story = memo(<WriteStory />);
+  return (
+    <>
+      Banner URL: <ImageGen clipLongPrompt>Generate an image for this story: {story}</ImageGen>
```

Comment:
Ah, I see why you wanted `clipLongPrompt` – so we could pass a `story` of arbitrary length and not have to worry about overflowing the context window.

I think putting a prop on `ImageGen` in particular is not the right approach. I think what would be better is to have a `<Summarize>{story}</Summarize>` component (perhaps similar to the demo @juberti added). We could have various summarizers with different strategies; in this case, we'd have a `TruncateSummarize` that simply truncates.

That way, we get this important use case (summarization / prompt length management) in a modular way that can be applied anywhere, not just on particular components.

Reply:
And in the meantime, if we just want to naively truncate the string (which I think is fine for now), but we also don't want to make a whole `Summarize` component with docs etc, I'd just write `{story.slice(0, maxLength)}`.

Reply:
agreed, clipping may or may not give you the desired result, better to make this explicit

Reply:
Yeah okay I'll remove it. The issue is that a separate summarizer would not know about the context length of the image model.

Reply:
> `{story.slice(0, maxLength)}`
Makes sense, but how do you know `maxLength`?

Reply:
Okay, what if we give the summarizer to the component itself?
```tsx
<ImageGen summarizer={<TruncateSummarizer />}>....<ImageGen>
```

This way the model can give summarizer any info that it needs. Also, it can decide not to summarize short responses.

Reply:
Where is Justin's example btw?

Reply:
I would say all this falls under the category of "new features", so let's defer it until after launch. For now, I'd say we should just remove this the `clipLongPrompt` prop and do a `slice` with `maxLength` set to something shorter than every context window, like `1000`. It'll be fine for the purposes of this demo.

I'd rather not pass `summarizer` as a prop to `ImageGen` for the same reasons stated above – structuring the prompt correctly is orthogonal to the `ImageGen` component. Other places will want this same functionality.

> The issue is that a separate summarizer would not know about the context length of the image model.

This is a good question. The summarizer would get it off `context`. We can talk more about how to do that after launch. :smile: